### ISSUE TITLE
bug: Sub Lambda container could not connect to DynamoDB

### ISSUE THREAD

**Issue by HenryHu-Testo** [localstack/localstack#10033]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I had a test case for the zipped nodejs Lambda function and I create the container for it by below code:
![image](https://github.com/localstack/localstack/assets/153203926/984c5228-adaa-4bb7-a0c2-57d1fc1db1c6)

I used aws-sdk v3 to setup the clients and I used `http://localhost:${localstack.getMappedPort(4566)}` for the endpoint. All the other test cases directly using the dynamodb client is fine, only that the test cases invoking the lambda to proceed the steps to call the DynamoDB with dynamodb client failed with below error:
`ERROR   Invoke Error    {"errorType":"Error","errorMessage":"connect ECONNREFUSED 127.0.0.1:44791","code":"ECONNREFUSED","errno":-111,"syscall":"connect","address":"127.0.0.1","port":44791,"$metadata":{"attempts":3,"totalRetryDelay":163},"stack":["Error: connect ECONNREFUSED 127.0.0.1:44791","    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)"]}`

There is also a simular case in [StackOverflow](https://stackoverflow.com/questions/77434611/unable-to-connect-localstack-dynamodb-from-lambda)

### Expected Behavior

The connection from Lambda container to DynamoDB becomes normal, or there could be a workaround at least

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    With TestContainers

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    NodeJs code invoke the Lambda with Lambda client to proceed the steps calling DynamoDB with Dynamodb client

### Environment

```markdown
- OS: Ubuntu 20.04.4
- LocalStack: 3.0.2
```

**Comment by dfangl** [comment]
Hi @HenryHu-Testo !
In general, if you want to reach LocalStack from your lambda, please use `localhost.localstack.cloud` instead of `localhost`. This is also outlined in our network troubleshooting guide: https://docs.localstack.cloud/references/network-troubleshooting/

`localhost` there will address the Lambda container itself, where dynamodb is not running.

Please report back if this resolves your issue!

**Comment by HenryHu-Testo** [comment]
> Hi @HenryHu-Testo ! In general, if you want to reach LocalStack from your lambda, please use `localhost.localstack.cloud` instead of `localhost`. This is also outlined in our network troubleshooting guide: https://docs.localstack.cloud/references/network-troubleshooting/
> 
> `localhost` there will address the Lambda container itself, where dynamodb is not running.
> 
> Please report back if this resolves your issue!

Yes, I could connect to the dynamodb with `localhost.localstack.cloud` andI also need to connect the lambda to my local service(not container), but now the network mode only allows `bridge`, I got confused with how to configure this. Could you please answer this also?

**Comment by dfangl** [comment]
In general, I would avoid setting both the network_mode and the `LAMBDA_DOCKER_NETWORK`, and it should still work. Please note that it has to run in a bridge network, `host` network mode for example is not supported.
If you need to reach your host, you can use `host.docker.internal` on macos, or in general `172.17.0.1` from linux (this mostly works, but special configurations of your docker client might prevent it from working).

**Comment by HenryHu-Testo** [comment]
> In general, I would avoid setting both the network_mode and the `LAMBDA_DOCKER_NETWORK`, and it should still work. Please note that it has to run in a bridge network, `host` network mode for example is not supported. If you need to reach your host, you can use `host.docker.internal` on macos, or in general `172.17.0.1` from linux (this mostly works, but special configurations of your docker client might prevent it from working).

Thank you, I think your answer resolve most of my problems.
